modules/db/src/main/scala/gem/dao/StepDao.scala

```diff
@@ -542,5 +542,118 @@ object StepDao {
             ${g.fpu.flatMap(_.toOption)})
         """.update
     }
+
+    object Gnirs {
+      import GnirsConfig.GnirsFpu
+
+      @SuppressWarnings(Array("org.wartremover.warts.Throw"))
+      final case class GnirsFpuBuilder(slit: Option[GnirsFpuSlit], other: Option[GnirsFpuOther]) {
+        def toFpu: GnirsFpu = (slit, other) match {
+          case (Some(s), None) => GnirsFpu.Slit(s)
+          case (None, Some(o)) => GnirsFpu.Other(o)
+          case _               => throw new IllegalArgumentException(s"GNIRS FPU must be either Slit or Other")
+        }
+      }
+
+      final case class GnirsBuilder(
+        camera:               GnirsCamera,
+        decker:               GnirsDecker,
+        disperser:            GnirsDisperser,
+        disperserOrder:       GnirsDisperserOrder,
+        exposureTime:         Duration,
+        filter:               GnirsFilter,
+        prism:                GnirsPrism,
+        fpuBuilder:           GnirsFpuBuilder,
+        readMode:             GnirsReadMode,
+        wavelengthSuggestion: GnirsWavelengthSuggestion
+      ) {
+        def toGnirs: DynamicConfig.Gnirs =
+          DynamicConfig.Gnirs(
+            camera,
+            decker,
+            disperser,
+            disperserOrder,
+            exposureTime,
+            filter,
+            fpuBuilder.toFpu,
+            prism,
+            readMode,
+            wavelengthSuggestion
+          )
+      }
```

Comment:
I'm guessing (maybe wrong) that we could define a composite for the FPU and then this `GnirsBuilder` wouldn't be necessary.  Did you run into problems with it?

Reply:
No, it was just straightforward enough with the builder instead of the composite.